akka-typed/src/main/scala/akka/typed/cluster/Cluster.scala

```diff
@@ -0,0 +1,256 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.cluster
+
+import akka.actor.{ Address, ExtendedActorSystem }
+import akka.annotation.{ DoNotInherit, InternalApi }
+import akka.cluster.ClusterEvent.{ ClusterDomainEvent, CurrentClusterState, MemberUp }
+import akka.cluster._
+import akka.pattern.FutureRef
+import akka.typed.internal.adapter.ActorSystemAdapter
+import akka.typed.scaladsl.Actor
+import akka.typed.{ ActorRef, ActorSystem, Extension, ExtensionId, Terminated }
+import akka.typed.scaladsl.adapter._
+import akka.util.Timeout
+
+import scala.concurrent.duration._
+import scala.collection.immutable
+import scala.concurrent.Await
+import scala.reflect.ClassTag
+
+/**
+ * Messages for subscribing to changes in the cluster state
+ *
+ * Not intended for user extension.
+ */
+@DoNotInherit
+sealed trait ClusterStateSubscription
+
+object Subscribe {
+  def apply[A <: ClusterDomainEvent: ClassTag](subscriber: ActorRef[A]) =
+    new Subscribe[A, A](subscriber, implicitly[ClassTag[A]].runtimeClass.asInstanceOf[Class[A]])
+}
+
+/**
+ * Subscribe to cluster state changes. The initial state of the cluster will be sent as
+ * a "replay" of the subscribed events.
+ *
+ * If all you are interested in is to react on this node becoming a part of the cluster,
+ * see [[OnSelfUp]]
+ *
+ * @param subscriber A subscriber that will receive events until it is unsubscribed or stops
+ * @param eventClass The type of events to subscribe to, can be individual event types such as
+ *                   `ReachabilityEvent` or one of the common supertypes, such as `MemberEvent` to get
+ *                   all the subtypes of events.
+ */
+final case class Subscribe[A <: ClusterDomainEvent, S <: A](
+  subscriber: ActorRef[A],
+  eventClass: Class[S]) extends ClusterStateSubscription
```

Comment:
`S` is needed? can't be just `A`?

Reply:
Wasn't sure about variance of Class and wanted to allow for example `Subscribe(ActorRef[MemberEvent], classOf[MemberUp])`

But it seems you are right, wasn't really needed.